akka-http-core/src/main/scala/akka/http/impl/engine/http2/client/ResponseParsing.scala

```diff
@@ -66,18 +66,24 @@ private[http2] object ResponseParsing {
         }
 
       } else remainingHeaders.head match {
-        case (":status", statusCodeValue) =>
+        case (":status", statusCodeValue: String) =>
           checkUniquePseudoHeader(":status", status)
           checkNoRegularHeadersBeforePseudoHeader(":status", seenRegularHeader)
           rec(remainingHeaders.tail, statusCodeValue.toInt, contentType, contentLength, seenRegularHeader, headers)
 
-        case ("content-type", ct) =>
+        case ("content-type", contentTypeValue: ContentType) =>
```

Comment:
Could it be worth matching both value types into one top level case and then match the type to cut one case it needs to check for every header?

Or perhaps even benefit from the hashcode/lookup-switch mentioned below if we _never_ matched on value in these? Maybe worth looking into in a separate PR.

Edit: Maybe that is exactly what you were saying in the comment above.

Reply:
> Edit: Maybe that is exactly what you were saying in the comment above.

Yep ;)